projects/wave-core-new/src/lib/operators/dialogs/point-in-polygon-filter/point-in-polygon-filter.component.ts

```diff
@@ -0,0 +1,90 @@
+import { Component, ChangeDetectionStrategy } from '@angular/core';
+
+import { RandomColorService } from '../../../util/services/random-color.service';
+
+import { Layer, VectorLayer } from '../../../layers/layer.model';
+import { ResultTypes } from '../../result-type.model';
+import {
+    AbstractVectorSymbology, PointSymbology,
+} from '../../../layers/symbology/symbology.model';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { WaveValidators } from '../../../util/form.validators';
+import { ProjectService } from '../../../project/project.service';
+import { zip } from 'rxjs';
+import { map, mergeMap } from 'rxjs/operators';
+import { WorkflowDict } from '../../../backend/backend.model';
+
+/**
+ * This component allows creating the point in polygon filter operator.
+ */
+@Component({
+    selector: 'wave-point-in-polygon-filter',
+    templateUrl: './point-in-polygon-filter.component.html',
+    styleUrls: ['./point-in-polygon-filter.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class PointInPolygonFilterOperatorComponent {
+
+    ResultTypes = ResultTypes;
+
+    form: FormGroup;
+
+    constructor(private randomColorService: RandomColorService,
+        private projectService: ProjectService,
+        private formBuilder: FormBuilder) {
+        this.form = formBuilder.group({
+            name: ['Filtered Values', [Validators.required, WaveValidators.notOnlyWhitespace]],
+            pointLayer: [undefined, Validators.required],
+            polygonLayer: [undefined, Validators.required],
+        });
+
+    }
+
+    add(event: Event) {
+        event.preventDefault(); // prevent page reload on error
+
+        if (this.form.invalid) {
+            return;
+        }
+
+        const points = this.form.controls['pointLayer'].value as Layer;
+        const polygons = this.form.controls['polygonLayer'].value as Layer;
+
+        const name: string = this.form.controls['name'].value;
+
+        // TODO: add projection operator
+        zip(
+            this.projectService.getWorkflow(points.workflowId),
+            this.projectService.getWorkflow(polygons.workflowId)
+        ).pipe(map(([a, b]) => {
```

Comment:
why do you use zip here and `forkJoin` on the other part?

Reply:
because there are only two inputs but potentially n layers to check? Idk shall I change it?